sdk/identity/azure-identity/tests/test_managed_identity_async.py

```diff
@@ -264,39 +279,60 @@ async def test_app_service_2017_09_01():
 async def test_app_service_user_assigned_identity():
     """App Service 2017-09-01: MSI_ENDPOINT, MSI_SECRET set"""
 
-    access_token = "****"
+    expected_token = "****"
     expires_on = 42
     client_id = "some-guid"
-    expected_token = AccessToken(access_token, expires_on)
     endpoint = "http://localhost:42/token"
     secret = "expected-secret"
     scope = "scope"
+    param_name, param_value = "foo", "bar"
+
     transport = async_validating_transport(
         requests=[
             Request(
                 base_url=endpoint,
                 method="GET",
                 required_headers={"secret": secret, "User-Agent": USER_AGENT},
                 required_params={"api-version": "2017-09-01", "clientid": client_id, "resource": scope},
-            )
+            ),
+            Request(
+                base_url=endpoint,
+                method="GET",
+                required_headers={"secret": secret, "User-Agent": USER_AGENT},
+                required_params={
+                    "api-version": "2017-09-01",
+                    "resource": scope,
+                    param_name: param_value,
+                },
+            ),
         ],
         responses=[
             mock_response(
                 json_payload={
-                    "access_token": access_token,
+                    "access_token": expected_token,
                     "expires_on": "01/01/1970 00:00:{} +00:00".format(expires_on),
                     "resource": scope,
                     "token_type": "Bearer",
                 }
             )
-        ],
+        ]
+        * 2,
     )
 
     with mock.patch(
-        "os.environ", {EnvironmentVariables.MSI_ENDPOINT: endpoint, EnvironmentVariables.MSI_SECRET: secret}
+        MANAGED_IDENTITY_ENVIRON, {EnvironmentVariables.MSI_ENDPOINT: endpoint, EnvironmentVariables.MSI_SECRET: secret}
```

Comment:
Is there a reason `clear=True` was added to the synchronous version of this test, but not to the async version here?

Reply:
Only my inattention 🙈